File: Prime List - GFG/prime-list.java
//{ Driver Code Starts
//Initial Template for Java
import java.io.*;
import java.util.*;

class Node{
    Node next;
    int val;
    public Node(int data){
        val=data;
        next=null;
    }
}

class Main
{
    public static void main(String args[])throws Exception
    {
        BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out=new PrintWriter(System.out);
        
        int t;
        t=Integer.parseInt(in.readLine());
        while(t-->0){
            int n;
            n=Integer.parseInt(in.readLine());
            Node head,tail;
            String s[]=in.readLine().trim().split(" ");
            int num=Integer.parseInt(s[0]);
            head=new Node(num);
            tail=head;
            for(int i=1;i<n;i++){
                num=Integer.parseInt(s[i]);
                tail.next=new Node(num);
                tail=tail.next;
            }
            Solution ob=new Solution();
            Node temp=ob.primeList(head);
            while(temp!=null){
                out.print(temp.val+" ");
                temp=temp.next;
            }
            out.println();
        }
        out.close();
    }
}
// } Driver Code Ends


//User function Template for Java
/*
class Node{
    Node next;
    int val;
    public Node(int data){
        val=data;
        next=null;
    }
}
*/

class Solution
{
    public static int near(boolean primes[], int n){
        boolean flag = true;
        if(primes[n]){
            return n;
        }
        int le = n - 1, ri = n+1, res = n;
        while(flag){
            if(primes[le]){
                res = le;
                flag = false;
                break;
            }
            if(primes[ri]){
                res = ri;
                flag = false;
                break;
            }
            le--;
            ri++;
        }
        return res;
    }
    Node primeList(Node head){
        //code here
        Node temp = head;
        boolean primes[] = new boolean[100001];
        Arrays.fill(primes, true);
        primes[0] = primes[1] = false;
        for(int i = 2 ; i*i <= 100000 ; i++){
            if(primes[i]){
                for(int j = i*i ; j <= 100000 ; j+=i){
                    primes[j] = false;
                }
            }
        }
        
        while(temp != null){
            temp.val = near(primes, temp.val);
            temp = temp.next;
        }
        return head;
    }
}

File: Prime List - GFG/prime-list.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

class Node{
public:
    int val;
    Node *next;
    Node(int num){
        val=num;
        next=NULL;
    }
};


// } Driver Code Ends
//User function Template for C++

/*

class Node{
public:
    int val;
    Node *next;
    Node(int num){
        val=num;
        next=NULL;
    }
};

*/

class Solution{
public:
    int near(vector<bool> primes, int n){
        int l = n-1, r = n+1, res = n;
        if(primes[n]){
            return res;
        }
        bool flag = true;
        while(flag){
            if(primes[l]){
                res = l;
                flag = false;
                break;
            }
            if(primes[r]){
                res = r;
                flag = false;
                break;
            }
            l--;
            r++;
        }
        return res;
    }
    Node *primeList(Node *head){
        vector<bool> primes(100000, true);
        primes[0] = primes[1] = false;
        for(int i = 2 ; i*i <= 100000 ; i++){
            if(primes[i]){
                for(int j = i*i ; j <= 100000 ; j+=i){
                    primes[j] = false;
                }
            }
        }
        Node * ptr = head;
        while(ptr != NULL){
            ptr->val = near(primes, ptr->val);
            ptr = ptr->next;
        }
        return head;
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        Node *head,*tail;
        int num;
        cin>>num;
        head=tail=new Node(num);
        for(int i=0;i<n-1;i++){
            int num;
            cin>>num;
            tail->next=new Node(num);
            tail=tail->next;
        }
        Solution ob;
        Node *temp=ob.primeList(head);
        while(temp!=NULL){
            cout<<temp->val<<" ";
            temp=temp->next;
        }
        cout<<endl;
    }
    return 0;
}
// } Driver Code Ends